src/ui/public/vislib/lib/axis.js

```diff
@@ -5,145 +5,113 @@ import ErrorHandlerProvider from 'ui/vislib/lib/_error_handler';
-      return this.scale.addInterval(interval);
+      return this.axisScale.addInterval(interval);
     }
 
     substractInterval(interval) {
-      return this.scale.substractInterval(interval);
+      return this.axisScale.substractInterval(interval);
     }
 
     tickScale(length) {
       const yTickScale = d3.scale.linear()
-        .clamp(true)
-        .domain([20, 40, 1000])
-        .range([0, 3, 11]);
+      .clamp(true)
+      .domain([20, 40, 1000])
+      .range([0, 3, 11]);
 
       return Math.ceil(yTickScale(length));
     }
 
     tickFormat() {
-      if (this.axisFormatter) return this.axisFormatter;
-      if (this.isPercentage()) return d3.format('%');
+      if (this.config.get('labels.axisFormatter')) return this.config.get('labels.axisFormatter');
+      if (this.config.isPercentage()) return d3.format('%');
       return d3.format('n');
     }
 
     getLength(el, n) {
-      if (this.isHorizontal()) {
-        return $(el).parent().width() / n - this._attr.margin.left - this._attr.margin.right - 50;
+      const margin = this.config.get('vis._attr.margin');
+      if (this.config.isHorizontal()) {
+        return $(el).parent().width() / n - margin.left - margin.right - 50;
       }
-      return $(el).parent().height() / n - this._attr.margin.top - this._attr.margin.bottom;
+      return $(el).parent().height() / n - margin.top - margin.bottom;
     }
 
     updateXaxisHeight() {
       const self = this;
-      const selection = d3.select(this.vis.el).selectAll('.vis-wrapper');
-
+      const position = this.config.get('position');
+      const selection = d3.select(this.config.get('rootEl')).selectAll('.vis-wrapper');
 
       selection.each(function () {
         const visEl = d3.select(this);
 
         if (visEl.select('.inner-spacer-block').node() === null) {
           visEl.selectAll('.y-axis-spacer-block')
-            .append('div')
-            .attr('class', 'inner-spacer-block');
+          .append('div')
+          .attr('class', 'inner-spacer-block');
         }
 
-        const height = visEl.select(`.axis-wrapper-${self.position}`).style('height');
-        visEl.selectAll(`.y-axis-spacer-block-${self.position} .inner-spacer-block`).style('height', height);
+        const height = visEl.select(`.axis-wrapper-${position}`).style('height');
+        visEl.selectAll(`.y-axis-spacer-block-${position} .inner-spacer-block`).style('height', height);
       });
     }
 
     adjustSize() {
       const self = this;
       const xAxisPadding = 15;
+      const style = this.config.get('style');
+      const margin = this.config.get('vis._attr.margin');
+      const position = this.config.get('position');
 
       return function (selection) {
         const text = selection.selectAll('.tick text');
         const lengths = [];
 
         text.each(function textWidths() {
           lengths.push((() => {
-            if (self.isHorizontal()) {
+            if (self.config.isHorizontal()) {
```

Comment:
i hate self as well ... mostly with arrow functions we should be able to get rid of it ,.... however d3 (as far as i know) is kind of dependand on the old functions ... maybe thats not the case anymore, i'll check it out